src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentContext.cs

```diff
@@ -138,9 +138,9 @@ public ValueTask<SourceText> GetHtmlSourceTextAsync(CancellationToken cancellati
             ? new(GetHtmlSourceTextCore(codeDocument))
             : GetHtmlSourceTextCoreAsync(cancellationToken);
 
-        static SourceText GetHtmlSourceTextCore(RazorCodeDocument codeDocument)
+        SourceText GetHtmlSourceTextCore(RazorCodeDocument codeDocument)
```

Comment:
Just add a `CancellationToken` parameter and leave the local function as static. Otherwise, this will unconditionally create a closure class at the start of the method to hold `cancellationToken` -- regardless of the result of `TryGetCodeDocument(...)`. (Many thanks to @ToddGrun for pointing that out to me awhile back.)